api-mock/mocks/ThirdPartyMock.ts

```diff
@@ -0,0 +1,75 @@
+import SuperTokens from "supertokens-node";
+import { User as UserClass } from "supertokens-node/lib/build/user";
+import ThirdParty from "supertokens-node/recipe/thirdparty";
+import { queryAPI } from "../fetcher";
+import { minify } from "../utils";
+
+export const ThirdPartyMock: Partial<typeof ThirdParty> = {
+    init: (config) => {
+        return {
+            config: JSON.stringify({
+                ...config,
+                ...(config?.override
+                    ? {
+                          override: {
+                              ...config.override,
+                              ...(config.override.apis
+                                  ? {
+                                        apis: minify(config.override.apis.toString()),
+                                    }
+                                  : {}),
+                          },
+                      }
+                    : {}),
+                ...(config?.signInAndUpFeature
+                    ? {
+                          signInAndUpFeature: {
+                              ...config?.signInAndUpFeature,
+                              ...(config?.signInAndUpFeature.providers
+                                  ? {
+                                        providers: config?.signInAndUpFeature.providers.map((p) => ({
+                                            ...p,
+                                            ...(p.override
+                                                ? {
+                                                      override: minify(p.override.toString()),
+                                                  }
+                                                : {}),
+                                        })),
+                                    }
+                                  : {}),
+                          },
+                      }
+                    : {}),
+            }),
+            recipeId: "thirdparty",
+        } as any;
+    },
+    manuallyCreateOrUpdateUser: async (
+        tenantId,
+        thirdPartyId,
+        thirdPartyUserId,
+        email,
+        isVerified,
+        session,
+        userContext
+    ) => {
+        const response = await queryAPI({
+            method: "post",
+            path: "/mock/thirdparty/manuallycreateorupdateuser",
+            input: { tenantId, thirdPartyId, thirdPartyUserId, email, isVerified, session, userContext },
```

Comment:
Doesn't the session object need any serialization?

Reply:
could be possible but it didn't affect any test, I was doing only changes to make the tests work, so is probably that we didn't have any request with the session defined

Reply:
there is a `handleSession` that handles deserialization in the API server. Is that used/tested?

Reply:
the session object is serialized by the JSON.stringify on the queryAPI function. 
handleSession is the helper to deserialize this stringified object. It is used in a few places like emailpassword, passwordless and thirdparty

Reply:
What I meant is if you tried the default serialization and `handleSession` working together as expected. It being in the codebase is no guarantee of that:
> so is probably that we didn't have any request with the session defined

Reply:
oh yes, is working, I only implement the handleSession because I need to make some tests work